arangod/Aql/VariableGenerator.cpp

```diff
@@ -65,21 +65,27 @@ std::unordered_map<VariableId, std::string const> VariableGenerator::variables(
 /// @brief generate a variable
 Variable* VariableGenerator::createVariable(std::string_view name,
                                             bool isUserDefined) {
-  std::string temp(name);
+  auto id = nextId();
+  auto nameCopy = [&] {
+    if (name.empty()) {
+      TRI_ASSERT(!isUserDefined);
+      return std::to_string(id);
+    } else {
+      return std::string{name};
+    }
+  }();
 
   if (isUserDefined && !isValidName(name.data(), name.data() + name.size())) {
     THROW_ARANGO_EXCEPTION_MESSAGE(
         TRI_ERROR_QUERY_PARSE,
         arangodb::basics::Exception::FillExceptionString(
-            TRI_ERROR_QUERY_VARIABLE_NAME_INVALID, temp.c_str()));
+            TRI_ERROR_QUERY_VARIABLE_NAME_INVALID, nameCopy.c_str()));
   }
```

Comment:
In case we run into this exception, we will needlessly have incremented the id value via a call to `nextId()`.
But we don't need the id value in case a user-defined name is provided.

So we would as well move the id increment call to below this check and thus avoid increasing it in case of obvious errors. This is not critical but may even make the resulting code simpler.